safekeeper/src/timeline.rs

```diff
@@ -684,11 +581,21 @@ impl Timeline {
     /// Get safekeeper info for broadcasting to broker and other peers.
     pub fn get_safekeeper_info(&self, conf: &SafeKeeperConf) -> SafekeeperTimelineInfo {
         let shared_state = self.write_shared_state();
-        shared_state.get_safekeeper_info(&self.ttid, conf)
+        shared_state.get_safekeeper_info(
+            &self.ttid,
+            conf,
+            self.walsenders.get_remote_consistent_lsn(),
+        )
     }
 
     /// Update timeline state with peer safekeeper data.
-    pub async fn record_safekeeper_info(&self, sk_info: &SafekeeperTimelineInfo) -> Result<()> {
+    pub async fn record_safekeeper_info(&self, sk_info: &mut SafekeeperTimelineInfo) -> Result<()> {
```

Comment:
Nothing added here needs `&mut` or am I wrong?

Reply:
No, its remote_consistent_lsn field is updated by taking max with local value.

Reply:
Maybe it's better to... 
```suggestion
    pub async fn record_safekeeper_info(&self, sk_info: mut SafekeeperTimelineInfo) -> Result<()> {
```

...to remove the reference to consume the value?

Reply:
changed